modules/core-layers/src/path-layer/path-layer.js

```diff
@@ -237,105 +273,95 @@ export default class PathLayer extends Layer {
     );
   }
 
-  calculateStartPositions(attribute) {
-    const {paths} = this.state;
-    const {value} = attribute;
-
-    let i = 0;
-    paths.forEach(path => {
-      const numSegments = path.length - 1;
-      for (let ptIndex = 0; ptIndex < numSegments; ptIndex++) {
-        const point = path[ptIndex];
-        value[i++] = point[0];
-        value[i++] = point[1];
-        value[i++] = point[2] || 0;
-      }
-    });
-  }
-
-  calculateEndPositions(attribute) {
-    const {paths} = this.state;
-    const {value} = attribute;
+  /**
+   * Positions and discardFlags
+   *
+   * For each path, the positions array is constructed as such:
+   * (vertex to the left of v0) v0 v1 v2 ... vn (vertex to the right of vn)
+   * We reuse the some positions buffer for all position attributes by offsetting:
+      instanceLeftPosition  (+0)
+      instanceStartPosition (+1)
+      instanceEndPosition   (+2)
+      instanceRightPosition (+3)
+   *
+   * `discardFlags` is used to mark the end of each path where vertices from the previous
+   * path overlap with vertices from the next path. We shouldn't draw these segments:
+   *
+      instanceLeftPosition    A0 A0 A1 A2 A3 A4 A4 B0 B0 B1 B2 B2 C0 ...
```

Comment:
Could the beginning zeros be cut just by making the buffers start with a bigger `offset`? Or this repeats per segment?